mne/viz/ica.py

```diff
@@ -1146,26 +1149,58 @@ def _plot_ica_overlay_raw(data, data_cln, times, title, ch_types_used, show):
-    assert data.shape == data_cln.shape
-    fig, (ax1, ax2) = plt.subplots(2, 1, sharex=True)
-    plt.suptitle(title)
-    ax1.plot(times, data.T, color="r")
-    ax1.plot(times, data_cln.T, color="k")
-    ax1.set(xlabel="Time (s)", xlim=times[[0, -1]], title="Raw data")
-
-    _ch_types = {"mag": "Magnetometers", "grad": "Gradiometers", "eeg": "EEG"}
-    ch_types = ", ".join([_ch_types[k] for k in ch_types_used])
-    ax2.set_title("Average across channels ({})".format(ch_types))
-    ax2.plot(times, data.mean(0), color="r")
-    ax2.plot(times, data_cln.mean(0), color="k")
-    ax2.set(xlabel="Time (s)", xlim=times[[0, -1]])
-    tight_layout(fig=fig)
+    ch_types = raw.get_channel_types(picks=picks, unique=True)
+    for ch_type in ch_types:
+        if ch_type in ("mag", "grad"):
+            fig, ax = plt.subplots(3, 1, sharex=True)
+        elif ch_type == "eeg" and not _has_eeg_average_ref_proj(raw.info):
+            fig, ax = plt.subplots(3, 1, sharex=True)
+        else:
+            fig, ax = plt.subplots(2, 1, sharex=True)
+        plt.suptitle(title)
+
+        # select sensors and retrieve data array
+        picks_by_type = _picks_to_idx(raw.info, ch_type, exclude=())
+        picks_ = np.intersect1d(picks, picks_by_type)
+        data, times = raw[picks_, start:stop]
+        data_cln, _ = raw_cln[picks_, start:stop]
+
+        # plot all sensors of the same type together
+        ax[0].plot(times, data.T, color="r")
+        ax[0].plot(times, data_cln.T, color="k")
+        _ch_type = (
+            DEFAULTS["titles"][ch_type] if ch_type in DEFAULTS["titles"] else ch_type
+        )
+        ax[0].set(xlabel="Time (s)", xlim=times[[0, -1]], title=f"Raw {_ch_type} data")
+
+        # second plot for M/EEG using GFP or RMS
+        if ch_type == "eeg":  # Global Field Power
+            ax[1].plot(times, np.std(data, axis=0), color="r")
+            ax[1].plot(times, np.std(data_cln, axis=0), color="k")
+            ax[1].set(
+                xlabel="Time (s)",
+                xlim=times[[0, -1]],
+                title=f"{_ch_type} Global Field Power",
+            )
+        elif ch_type in ("mag", "grad"):  # RMS
+            ax[1].plot(times, np.sqrt((data**2).mean(axis=0)), color="r")
+            ax[1].plot(times, np.sqrt((data_cln**2).mean(axis=0)), color="k")
+            ax[1].set(xlabel="Time (s)", xlim=times[[0, -1]], title=f"{_ch_type} RMS")
+
+        # last plot with the average across all channels of the same type
+        if ch_type != "eeg" or not _has_eeg_average_ref_proj(raw.info):
+            ax[-1].plot(times, data.mean(axis=0), color="r")
+            ax[-1].plot(times, data_cln.mean(axis=0), color="k")
+            ax[-1].set(
+                xlabel="Time (s)",
+                xlim=times[[0, -1]],
+                title=f"Average across {_ch_type} channels",
+            )
 
-    fig.subplots_adjust(top=0.90)
-    fig.canvas.draw()
+        # figure format
+        tight_layout(fig=fig)
```

Comment:
do we need this?  In general I'm trying to move away from tight_layout and use `constrained_layout` when possible (or neither, if it looks OK).

Reply:
I don't have any preference, this part is re-used from the old function.
I moved to a constrained layout, looks good as well. It has a bit more space between the titles and a bit less space on the left and right side of the subplot.

Reply:
Ok, I must have misread the diff. I thought you were adding it. Ok with me to leave it in, we can switch to constrained layout later in a PR that removes all tight_layouts at once

Reply:
Well I already changed to a constrained layout here since you preferred that one. Both look OK anyway, I think we can leave the constrained layout.

Reply:
OK fine by me.  I didn't notice before that you'd already changed to constrained_layout, because I was responding on a phone.